### ISSUE TITLE
Virtual bus parsing bug for verilog 

### ISSUE THREAD

**Issue by SebastianHambura** [TerosTechnology/vscode-terosHDL#314]
Hi, 

I've just discovered your tool, and it looks very good ! I have some troubles getting the virtualbus feature to work though. It feel like the parser is not working as intended for Verilog :

**Case 1** : Just not working
```
module test(
    
    //! @virtualbus test
    input a, 
    input b
    //! @end

);  
endmodule
```
![grafik](https://user-images.githubusercontent.com/73603223/164012768-112c0da5-5f38-4357-89c9-3c8c337a23b2.png)

**Case 2** : including a port which shouldn't be in the virtual bus
```
module test(
    
    //! @virtualbus test
    input a, 
    input b, 
    //! @end

    input not_in_bus
);    
endmodule 
```
![grafik](https://user-images.githubusercontent.com/73603223/164012526-90ecee7a-c4c9-4b5c-bcf7-613dcec81a0e.png)

**Case 3** : adding inline comment, now the virtual bus is missing again
```
module test(
    
    //! @virtualbus test
    input a, 
    input b,
    //! @end

    input not_in_bus //! not anymore

);    
endmodule
```
![grafik](https://user-images.githubusercontent.com/73603223/164013108-64d0fb9b-d2e0-446d-9fb2-3a0709357232.png)

**Case 4**: but with multiline comments it goes back to case 2
```
module test(
    
    //! @virtualbus test
    input a, 
    input b,
    //! @end

    //! not anymore
    input not_in_bus 

);    
endmodule
```
![grafik](https://user-images.githubusercontent.com/73603223/164013374-c90d2a83-8f13-4e7f-b862-05e533f23d19.png)

**Case 5**: working 
```
module test(
    
    //! @virtualbus test
    input a, 
    input b, //! @end
    

    //! not anymore
    input not_in_bus 

);    
endmodule
```
![grafik](https://user-images.githubusercontent.com/73603223/164014037-175bc282-df9c-4855-8715-cd0ae470e8ff.png)

**Case 6**: end above the last port of the virtual bus works correctly
```
module test(
    
    //! @virtualbus test
    input a,
    //! @end 
    input b,  

    
    input not_in_bus //! not anymore

);    
endmodule

```
![grafik](https://user-images.githubusercontent.com/73603223/164014940-ad8e3113-c7b3-456b-87d6-b213374ed9ef.png)

**TL;DR** : the "end" tag seems to be grouped with the next following port, instead of the port right above it. This leads to problems if it's at the end (no port following it, so it gets dropped), or it can have wanky interactions with the next port (including an additional port into the virtualbus, not working at all, etc...). A solution is to 'force' the tag to be grouped with the last port of the virtualbus, either by putting it as inline comment (so to the right) or as multiline comment (so just above). This last solution is not nice visually, and don't follow the documentation : "All the ports between the lines: --! @virtualbus and --! @end will be grouped in the diagram and the ports table"

 - OS: Windows 11
 - VSCode version 1.66.2
 - TerosHDL version : v2.0.6 (freshly installed from VSCode extension)

**Comment by qarlosalberto** [comment]
Thanks for the long explanation!!

Really this not a bug. A comment if associated to a port if it's inline or above the declaration. Here:

```
module test(
    
    //! @virtualbus test
    input a, 
    input b
    //! @end

);  
endmodule
```

"@end" isn't associated to any port. So you need to associate it to input b


```
module test(
    
    //! @virtualbus test
    input a, 
    input b //! @end
);  
endmodule
```

or


```
module test(
    
    //! @virtualbus test
    input a, 
    //! @end
    input b 
);  
endmodule
```

But your are right, it's a little confused. So I will change the label from bug to  enhancement .

**Comment by SebastianHambura** [comment]
In that case, could you maybe also check if the same applies to VHDL ? Because the example code given in the documentation (https://terostechnology.github.io/terosHDLdoc/documenter/start.html) imply you can write the `--! @end` tag below the last port of the virtual bus. If it's a Verilog vs VHDL parsing difference, it would be very cool to have a page with all these subtilities :)

Also, the sample code is not producing the documentation as expected : there are 3 `@virtualbus` but only 2 `@end` (I guess there is one missing line 65), mem_dout is not drawn on the diagram and only 2 virtual busses are created in the documentation (even after adding the missing `@end`).

That being said, thanks a lot for this awesome tool ! :D

**Comment by qarlosalberto** [comment]
You are right. Quite a while ago I made some changes, but honestly I don't remember them...
I will check it and I will improve the documentation.

Between "@virtualbus" you don't need to declare the "@end". E.g:

```
module test(
    
    //! @virtualbus test0
    input a, 
    input b,
    //! @virtualbus test1
    input c,
    input d //! @end
);  
endmodule
```

![image](https://user-images.githubusercontent.com/5955143/164074127-e88c2e51-9197-473a-9edd-7195a47f21ca.png)

**Comment by Luca-s-Works** [comment]
Hello!
First of all ty for all the hard work done with this VS plugin.

Nevertheless though, I'd like to reopen the discussion on this "feature".

Suppose I want to add @virtualbus information to an entity/module with this commenting style (inline to port declaration):
```
entity test is
    port(
        test0_a     : std_logic;    --! Port A.
        test0_b     : std_logic;    --! Port B.

        sys_clk     : std_logic;    --! Clock reference signal.
        sys_arstn   : std_logic     --! Async reset signal (active_low).
    );
end test;
```

Where am I supposed to place the @end for the @virtualbus?

None of these works:

```
        --! @virtualbus test0
        test0_a     : std_logic;    --! Port A.
        test0_b     : std_logic;    --! Port B.
        --! @end
```

```
        --! @virtualbus test0
        test0_a     : std_logic;    --! Port A.
        test0_b     : std_logic;    --! Port B. @end
```

```
        --! @virtualbus test0
        test0_a     : std_logic;    --! Port A.
        test0_b     : std_logic;    --! @end Port B.
```

```
        --! @virtualbus test0
        test0_a     : std_logic;    --! Port A.
        test0_b     : std_logic;    --! Port B. --! @end
```
